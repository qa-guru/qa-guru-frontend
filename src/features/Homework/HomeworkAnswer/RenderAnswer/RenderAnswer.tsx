import React from "react";
import { Box, Link, Typography } from "@mui/material";
import { IRenderAnswer } from "./RenderAnswer.types";

const RenderAnswer: React.FC<IRenderAnswer> = ({ answer }) => {
  const linkRegex = /(https?:\/\/[^\s]+)/g;
  const lines = answer.split("\n");

  return (
    <Box mt="16px">
      {lines.map((line, i) => {
        const parts = line.split(linkRegex);
        return (
          <Typography variant="subtitle1" key={i}>
            {parts.map((part, j) =>
              linkRegex.test(part) ? (
                <Link key={j} href={part}>
                  {part}
                </Link>
              ) : (
                <React.Fragment key={j}>{part}</React.Fragment>
              )
            )}
            <br />
          </Typography>
        );
      })}
    </Box>
  );
};

export default RenderAnswer;
