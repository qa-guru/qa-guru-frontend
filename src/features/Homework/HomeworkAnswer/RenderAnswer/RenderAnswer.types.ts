export interface IRenderAnswer {
  answer: string;
}
