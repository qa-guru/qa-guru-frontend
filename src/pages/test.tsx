import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
