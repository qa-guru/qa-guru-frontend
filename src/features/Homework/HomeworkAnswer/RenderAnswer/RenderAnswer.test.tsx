import { MemoryRouter } from "react-router-dom";
import { render } from "../../../../test/utilsTest";
import RenderAnswer from "./RenderAnswer";

describe("HomeworkAnswer", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <RenderAnswer answer={"answer"} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
