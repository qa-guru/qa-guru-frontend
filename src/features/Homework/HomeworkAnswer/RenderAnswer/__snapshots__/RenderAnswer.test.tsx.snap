// Vitest Snapshot v1

exports[`HomeworkAnswer > the component is render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1yuhvjn"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
    >
      answer
      <br />
    </h6>
  </div>
</DocumentFragment>
`;
